templates/zerver/help/rename-a-topic.md

```diff
@@ -54,6 +54,12 @@ another.
 
 1. Click **Move topic**.
 
+
+!!! warn ""
+    **Note**: When a topic is moved to a private stream with protected history,
+              messages in the topic will be visible to all the subscribers.
```

Comment:
What happens when the old stream previously had protected history and the new stream does too?  I feel like as a user I'd expect that arrangement to be preserved?  Not sure.

Reply:
Should I add an option for user to check/uncheck whether the messages should be visible to all subs in the new stream? I feel like this is something person moving the messages will want to configure.

Reply:
Yeah, I think that's probably the right solution here.  Maybe we can call it:

'Add messages to personal history of subscribers to #newstream'?

Reply:
(And perhaps we should make the behavior of creating UserMessage elements conditional on the recipients needing them, i.e. it being a protected history stream we're moving to?)

Reply:
Okay, I will do a followup on it.